openwisp_firmware_upgrader/tests/test_models.py

```diff
@@ -31,6 +32,21 @@ class TestModels(TestUpgraderMixin, TestCase):
     app_label = 'openwisp_firmware_upgrader'
     os = 'OpenWrt 19.07-SNAPSHOT r11061-6ffd4d8a4d'
     image_type = REVERSE_FIRMWARE_IMAGE_MAP['YunCore XD3200']
+    MAX_FILENAME_LENGTH = 255
+
+    def test_filename_within_limit(self, **kwargs):
+        org = kwargs.pop('organization', self._get_org())
+        category = kwargs.pop('category', self._get_category(organization=org))
+        build1 = self._create_build(category=category, version='0.1')
+        validlength = self.MAX_FILENAME_LENGTH
+        file = SimpleUploadedFile('a' * validlength, b'')
+        image = self._create_firmware_image(
+            build=build1, type=self.TPLINK_4300_IMAGE, file=file
+        )
+        try:
+            image.full_clean()
+        except ValidationError:
+            self.fail("Filename length exceeds the limit of 255 characters")
```

Comment:
As discussed, creating a file with a name longer than 255 chars on Linux is not possible.
Moreover, we trust that the django model logic which caps the max length of a field works, so we don't need to test it again.